### PR TITLE
Upgrade to node 20

### PR DESCRIPTION
**Pull request by alessio-perugini**
GitHub will deprecate the node runtime soon. ( https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/)
We're bumping the node version to 20.
After merging this, must follow a bump of the major version of this action.

### CODE REVIEW

File: .github/workflows/check-npm-task.yml [hunk 1]
```diff
@@ -73,8 +73,9 @@ jobs:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
           version: 3.x
 
-      - name: Validate package.json
-        run: task --silent npm:validate
+      # Temprary disable as ajv seems to have problems
+      #- name: Validate package.json
+      #  run: task --silent npm:validate
```
Comment (per1234) [inline .github/workflows/check-npm-task.yml:78]: There is now a fix for the problem with the `npm:validate` task in the upstream "template":

https://github.com/arduino/tooling-project-assets/commit/cc6f87c1bf8cb188d5b4f258900531c4e67d301b

**Review by per1234 — CHANGES_REQUESTED** [review]

**Review by per1234 — APPROVED** [review]
Thanks Alessio!

**Review by MatteoPologruto — APPROVED** [review]

**Comment by kayabaNerve** [comment]
The CI on the commit merging this appears to have failed, despite the CI here passing.

https://github.com/arduino/setup-protoc/actions/runs/7728930562/job/21070931021

Is any context/insight available on this?

**Comment by per1234** [comment]
@kayabaNerve the failure was caused by an external change to the `package.json` schema that was made between the time of the CI run for this PR and when the PR was merged.

The cause of the workflow run failure was only a defect in the project's CI system, not in the content of the project. So the workflow run failure it is nothing for users of the action to be concerned about. 

I submitted a fix with details about the problem here: https://github.com/arduino/setup-protoc/pull/96

**Pull request merged**